### PR TITLE
fix: Handle zstd encoding in webpack proxy config

### PR DESCRIPTION
**Pull request by kgabryje**
### SUMMARY
Some of the latest changes caused the `localhost:9000` to use zstd encoding, which we didn't handle in `webpack.proxy-config.js`. This PR adds `simple-zstd` lib to decompress zstd encoded data.

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF
Before:

image

After:

image

### TESTING INSTRUCTIONS
Run `npm run dev-server`, go to `localhost:9000`, verify that it works normally

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

**Review by geido — APPROVED** [review]

**Review by rusackas — APPROVED** [review]

**Comment by michael-s-molina** [comment]
THANK YOU!

**Comment by villebro** [comment]
Also big thanks from me! I'd be curious to understand what caused this regression..

**Comment by rusackas** [comment]
Closing https://github.com/apache/superset/pull/29822 as superseded by this one.

File: superset-frontend/package.json [hunk 1]
```diff
@@ -201,6 +201,7 @@
     "rimraf": "^6.0.1",
     "rison": "^0.1.1",
     "scroll-into-view-if-needed": "^3.1.0",
+    "simple-zstd": "^1.4.2",
```
Comment (mistercrunch) [inline superset-frontend/package.json:204]: @kgabryje shouldn't this be in `devDependencies`

**Comment by mistercrunch** [comment]
Having issues on master RN, saying `zstd` not installed. Though npmjs.org seems to be having issues, with `npm ci --verbose` stalling for minutes at a time...

File: superset-frontend/package.json (hunk 1 above)
Comment (kgabryje) [reply superset-frontend/package.json:204]: It should, I moved it to dependencies to silence eslint errors. It shouldn't really affect the entry point sizes though, as webpack will put this package in a chunk that's not downloaded on any page

**Comment by kgabryje** [comment]
> Having issues on master RN, saying `zstd` not installed. Though npmjs.org seems to be having issues, with `npm ci --verbose` stalling for minutes at a time...
> 
> In `docker-compose up`
> 
> ```
> superset_node         | [webpack-cli] Failed to load '/app/superset-frontend/webpack.config.js' config
> superset_node         | [webpack-cli] Error: Can not access zstd! Is it installed?
> ```

I put `apt-get install zstd` in `Dockerfile` and it works on CI build. Do you have an idea why it's not available in docker compose?

**Comment by mistercrunch** [comment]
I don't understand why it's not installed while it's specified here https://github.com/apache/superset/blob/master/Dockerfile#L35, debugging RN and had a `test.js` in the docker, didn't work, ran `apt-get install zstd` while in the container and it worked after. Very confusing.

**Comment by kgabryje** [comment]
> I don't understand why it's not installed while it's specified here https://github.com/apache/superset/blob/master/Dockerfile#L35, debugging RN and had a `test.js` in the docker, didn't work, ran `apt-get install zstd` while in the container and it worked after. Very confusing.

Is it possible that docker compose skips the part where we install zstd in existing containers?

**Comment by mistercrunch** [comment]
idk - was messing around with it and now exact same SHA command and now magically works, feels like there was a corrupt binary somewhere at Debian and now it's fixed.

**Comment by MialLewis** [comment]
> idk - was messing around with it and now exact same SHA command and now magically works, feels like there was a corrupt binary somewhere at Debian and now it's fixed.

This is happening to me... any idea what the magic fix could have been?

**Comment by MialLewis** [comment]
I managed to fix it by hacking in `"dev": "apt-get update && apt-get install zstd && webpack --mode=development --color --watch"`

here: https://github.com/apache/superset/blob/master/superset-frontend/package.json#L49

**Comment by mistercrunch** [comment]
Really shouldn't be problem, wondering if nuking docker caches and `docker compose build` should get you back on your feet.

**Pull request merged**